src/objects/Mesh.d.ts

```diff
@@ -1,19 +1,20 @@
 import { Geometry } from './../core/Geometry';
 import { Material } from './../materials/Material';
+import { MeshBasicMaterial } from './../materials/MeshBasicMaterial';
 import { Raycaster } from './../core/Raycaster';
 import { Object3D } from './../core/Object3D';
 import { BufferGeometry } from '../core/BufferGeometry';
 import { Intersection } from '../core/Raycaster';
 
-export class Mesh extends Object3D {
+export class Mesh<
+	TGeometry extends Geometry | BufferGeometry = BufferGeometry,
+	TMaterial extends Material | Material[] = MeshBasicMaterial
```

Comment:
Why did you set the default type to `MeshBasicMaterial` instead of `Material | Material[]` ?
It will fail on most of existing TypeScript codes when we use `Mesh` for interfaces or types.

I assume you were intended to indicate the type of default constructors, and it works fine when we just do `new Mesh()` .
But we have to replace every existing `Mesh` in interfaces with `Mesh<THREE.Geometry | THREE.BufferGeometry, THREE.Material | THREE.Material[]>` (or something strict) 😩 
I'm not sure whether this is an acceptable change or not.

Reply:
@Mugen87 how do you think about it?

Reply:
Sorry, I'm not into TS enough to review this code.

Reply:
How about this type declarations?

```ts
interface Mesh<
    TGeometry extends Geometry | BufferGeometry = Geometry | BufferGeometry,
    TMaterial extends Material | Material[] = Material | Material[],
> extends Object3D {
    geometry: TGeometry;
    material: TMaterial;

    // ...
}

declare var Mesh: {
    prototype: Mesh;
    new<
        TGeometry extends Geometry | BufferGeometry = BufferGeometry,
        TMaterial extends Material | Material[] = MeshBasicMaterial,
    >(geometry?: TGeometry, material?: TMaterial): Mesh<TGeometry, TMaterial>;
}
```

Reply:
My coworker gave me a great suggestion!! He says he has referenced one of dom.d.ts declaration.

Reply:
I just have confirmed this works pretty well on my existing codes. @jedahan could you check this out

Reply:
Sorry, been busy. Will likely be able to get to this tomorrow evening.